cmd/sonobuoy/app/args.go

```diff
@@ -401,3 +391,76 @@ func AddExtractFlag(flag *bool, flags *pflag.FlagSet) {
 		"If true, extracts the results instead of just downloading the results",
 	)
 }
+
+type e2eFocusFlagType PluginEnvVars
```

Comment:
So most of the 'added' lines are like these: we remove logical complexity throughout the codebase and for this flag-specific stuff, wire up these flag types. The flag methods of get/set/etc handle all the complexity in a very narrowly scoped point in code/time.

So in these cases, each of the focus/skip/parallel/mode, you get a new flag type which defines a custom action when you set it. These end up setting env vars but you could imagine them doing other things if needed.